backend/src/routes/payments.js

```diff
@@ -0,0 +1,58 @@
+const express = require('express');
+const services = require('../services');
+const router = express.Router();
+const db = require('../db');
+const auth = require('../auth');
+const utils = require('../utils');
+
+const getPaymentRedirectUrl = async (req, res) => {
+    try {
+        const paymentObj = req.body;
+        const validationErrors = utils.payments.validateAmount(
+            paymentObj.amount
+        );
+        if (validationErrors) {
+            res.status(422).json(validationErrors);
+        } else {
+            const user = req.user;
+            const dbUser = await db.users.getUser(user.phoneNumber);
+            paymentObj.userId = dbUser.id;
+            paymentObj.username = dbUser.username;
+            const url = await services.payments.getPaymentRedirectUrl(
+                paymentObj
+            );
+            res.status(200).json(url);
+        }
+    } catch (err) {
+        res
+            .status(500)
+            .json(`Failed to get payment redirect url. Error: ${err.message}`);
+    }
+};
+
+const paymentNotify = async (req, res) => {
+    res.status(200).json('Payment notification request received from Bambora');
+};
+
+const paymentReturn = async (req, res) => {
+    const response = req.query.RETURN_CODE;
+    if (response === 0) {
+        const ordernumber = req.query.ORDER_NUMBER;
+        const payment = await db.payments.getPayment(ordernumber);
```

Comment:
Would be good to explicitly handle the case where the payment is not found in our system and log it.